OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -159,21 +159,19 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 					// Try to find an alternative landing spot if we can't land at the current destination
 					if (!aircraft.CanLand(destination) && dropRange > 0)
 					{
-						var mobiles = cargo != null ? cargo.Passengers.Select(a =>
-						{
-							var mobileInfo = a.Info.TraitInfoOrDefault<MobileInfo>();
-							if (mobileInfo == null)
-								return new Pair<MobileInfo, uint>(null, 0);
-
-							return new Pair<MobileInfo, uint>(mobileInfo, (uint)mobileInfo.GetMovementClass(a.World.Map.Rules.TileSet));
-						}) : new Pair<MobileInfo, uint>[0];
+						var locomotors = cargo.Passengers
+							.Select(a => a.Info.TraitInfo<MobileInfo>())
```

Comment:
This should be a `TraitInfoOrDefault` - we then filter out any nulls on the next line.